apps/dotcom/src/components/SnapshotsEditor.tsx

```diff
@@ -58,6 +58,9 @@ interface SnapshotEditorProps {
 }
 
 export function SnapshotsEditor({ schema, records }: SnapshotEditorProps) {
+	// make sure this runs before the editor is instantiated
+	useState(convertLegacyUrlParams)
```

Comment:
should this maybe be a `useLayoutEffect`? Or just calling it directly? feels very odd to use `useState` to run an effect in this way.

alternatively, could we implement this via the `getUrl` API?

Reply:
switched this to a layoutEffect. good call